app/frontend/wallet/shelley-wallet.ts

```diff
@@ -204,13 +204,14 @@ const ShelleyWallet = ({
     return cryptoProvider.isHwWallet()
   }
 
-  function getHwWalletName() {
-    return isHwWallet ? (cryptoProvider as any).getHwWalletName() : undefined
+  function getWalletName() {
+    return cryptoProvider.getWalletName()
   }
 
   function submitTx(signedTx): Promise<any> {
+    const params = {walletType: getWalletName()}
```

Comment:
Do you think we could also pass user's staking key? It would allow much easier support

Reply:
I ll make this in another PR :+1: